### ISSUE TITLE
Rule proposal: the inverse of prefer-dom-node-dataset

### ISSUE THREAD

**Issue by bschlenk** [sindresorhus/eslint-plugin-unicorn#2451]
### Description

I work in a large codebase that happens to use lots of magic data attributes to change behavior of things in subtle ways. It can hurt greppability when seeing one of these data attributes used in css, but on the js side the data attribute is accessed via the `dataset` property. If you search the codebase for the `data-*` attribute you see in css, you won't find its usage on the js side and you have to know to strip the data and camelCase the name to find it.

It'd be nice if there was an inverse of `prefer-dom-node-dataset`, or if it took an option to invert the check.

Maybe this is easy enough to set up with [no-restricted-syntax](https://eslint.org/docs/latest/rules/no-restricted-syntax)?

### Fail

```js
el.dataset.someLongDataAttribute
```

### Pass

```js
el.hasAttribute('data-some-long-data-attribute')
```

### Proposed rule name

prefer-dom-node-has-attribute

**Comment by fregante** [comment]
I had the same experience with this rule: I disable it because it breaks greppability between JS and HTML(/JSX)
